### DISCUSSION TITLE
architecture ignored on macOS?

### DISCUSSION THREAD

**Discussion by bhundt** [premake/premake-core#2123] [Q&A]
Hey everybody, 

I am quite new to premake so maybe I am doing something wrong. I am working on an M1 Mac running 13.3.1. I want to cross-compile to a target architecture of x86_64 and for that I tried the sample premake project and adapted the premake file as follows:

```
workspace "HelloWorld"
   configurations { "Debug", "Release" }
   architecture "x86_64"

project "HelloWorld"
   kind "ConsoleApp"
   language "C++"
   cppdialect "C++20"
   targetdir "bin/%{cfg.buildcfg}"

   files { "**.hpp", "**.h", "**.cpp" }

   filter "configurations:Debug"
      defines { "DEBUG" }
      symbols "On"

   filter "configurations:Release"
      defines { "NDEBUG" }
      optimize "Speed"
```

Unfortunately this creates the default architecture of ```arm64``` which I checked using the `lipo -info` command.

What am I missing here? I tried using gmake as well as xcode4 and I am running 5.0.0-beta2.

Thanks and cheers
Bastian

**Comment by redorav** [comment]
Could you try architecture("x64") instead?

**Comment by bhundt** [comment]
Hey @redorav, unfortunately it still creates `arm64` when setting architecture to "x64"

**Comment by nickclark2016** [comment]
At this point, it may be worth at least opening a bug/issue for tracking, rather than through a discussion.
